src/components/CardExpirationSection/CardExpirationSection.tsx

```diff
@@ -1,61 +1,51 @@
 import styles from './CardExpirationSection.module.css';
 import { InputSection } from '../InputSection/InputSection';
-import { Dispatch, SetStateAction, useState } from 'react';
+import Input from '../Input/Input';
+import { ExpirationKey, ExpirationType } from '../../types';
 
-type Props = {
-  expiration: string[];
-  setExpiration: Dispatch<SetStateAction<string[]>>;
+type CardExpirationSectionProps = {
+  expiration: ExpirationType;
+  expirationRef: { month: React.RefObject<HTMLInputElement | null>; year: React.RefObject<HTMLInputElement | null> };
+  handleExpirationChange: (field: ExpirationKey, value: string) => void;
 };
-
-export default function CardExpirationSection({ expiration, setExpiration }: Props) {
-  const [expirationError, setExpirationError] = useState<string[]>(['', '']);
-
-  const handleExpirationChange = (index: number, value: string) => {
-    let errorMsg = '';
-    if (!/^[0-9]*$/.test(value)) {
-      errorMsg = '숫자만 입력 가능합니다.';
-      return;
-    }
-
-    if (index === 0) {
-      if (value !== '') {
-        const month = Number(value);
-        if (month < 1 || month > 12) {
-          errorMsg = '1부터 12 사이의 숫자를 입력해주세요.';
-        }
-      }
-    }
-    if (index === 1) {
-      if (value !== '' && value.length !== 2) {
-        errorMsg = '2자리 숫자를 입력해주세요.';
-      }
-    }
-
-    const updatedExpiration = [...expiration];
-    updatedExpiration[index] = value;
-    setExpiration(updatedExpiration);
-
-    const updatedError = [...expirationError];
-    updatedError[index] = errorMsg;
-    setExpirationError(updatedError);
-  };
+export default function CardExpirationSection({ expiration, expirationRef, handleExpirationChange }: CardExpirationSectionProps) {
+  const isError = expiration.month.errorMessage || expiration.year.errorMessage;
```

Comment:
```suggestion
  const errorMessage = expiration.month.errorMessage || expiration.year.errorMessage;
  const isError = Boolean(errorMessage)
```
`expiration.month.errorMessage || expiration.year.errorMessage`로직은 아래에서 재사용되기도 하고, `isError`는 Boolean 타입을 가질 것으로 예상되기 때문에 변수를 2개로 나눠도 괜찮을 것 같아요!(지금 동작이랑 크게 다른건 없어서 취향 차이일 수도 있겠네요🙃)

Reply:
넘 감사합니다!